### PR TITLE
feat: add setVariables function

### LINKED ISSUE #111: feat request: edit multiple variables at once
**Is your feature request related to a problem? Please describe.**
If someone wants to edit more than one variable using a mock contract, the user would have to repeat using the `setVariable` function. I believe it would help if there was a function that would edit multiple variables at once (like smock's v1 `smodify.put`)

### PR DESCRIPTION
**Pull request by tonykogias**
Adding a new function for mock contracts. With setVariables mocks will
be able to edit multiple internal variables at once.

- Fixes #111

### CODE REVIEW

File: src/logic/editable-storage-logic.ts [hunk 1]
```diff
@@ -22,4 +22,18 @@ export class EditableStorageLogic {
...
     }
   }
+
+  async setVariables(variables: any = {}) {
+    if (variables === undefined || variables === null) return;
+
+    let slots: any[] = [];
+
+    for (const [variableName, variableValue] of Object.entries(variables)) {
+      slots = slots.concat(computeStorageSlots(this.storageLayout, { [variableName]: variableValue }));
```
Comment (smartcontracts) [inline src/logic/editable-storage-logic.ts:32]: Alternatively maybe we could do `await setVariable(variableName, variableValue)`.

Comment (tonykogias) [reply src/logic/editable-storage-logic.ts]: Thanks! I pushed the fix.

**Comment by sambacha** [comment]
any chance of this being accepted?

File: test/unit/mock/editable-storage-logic.spec.ts [hunk 2]
```diff
@@ -17,130 +17,211 @@ describe('Mock: Editable storage logic', () => {
...
+      const mapValue = 5678;
+      await mock.setVariables({
+        _uint256Map: {
+          [mapKey]: mapValue
+        },
+        _uint256NestedMap: {
+          [mapKey]: {
+            [mapKeyB]: mapValue
+          }
+        }
```
Comment (wei3erHase) [inline test/unit/mock/editable-storage-logic.spec.ts:206]: this is gorgeous 😍, would you add some multiple inputs into the maps?

File: src/logic/editable-storage-logic.ts [hunk 3]
```diff
@@ -22,4 +22,14 @@ export class EditableStorageLogic {
       await this.vmManager.putContractStorage(toFancyAddress(this.contractAddress), fromHexString(slot.key), fromHexString(slot.val));
     }
   }
+
+  async setVariables(variables: any = {}) {
```
Comment (wei3erHase) [inline src/logic/editable-storage-logic.ts:26]: i'd love to see a type enforcement here

**Review by wei3erHase — CHANGES_REQUESTED** [review]
yo! @tonykogias THANKS for this contribution 🚀 
this looks like an amazing feature, very keen to push it forward, will help to organize my tests A LOT, data settings being stored in a couple lines `<3`

as for the tests, seem quite complete, and requesting to test all the different combinations is not possible, i'd just add on the test an example showing multiple inputs to the same variable

please think of typing to the input of the function to document what's expected

and always remember to `yarn lint:fix` 🙏  (will try to automate it soon)

{\\__/}
( • . •)
/ > `<3`

**Comment by wei3erHase** [comment]
yo @tonykogias, there's an active [bounty on this task](https://app.dework.xyz/i/5j1hLBSDZzKc19ZMGOm9o3) and is assigned to you, so that other devs don't overlap your work.
please confirm if you'll be continue working on it so that we keep that assignment,
again, thanks for your contribution! `<3`

**Comment by tonykogias** [comment]
I will resolve the comments later today

**Comment by tonykogias** [comment]
@wei3erHase I pushed the fixes please let me know if the changes work for you.

File: test/unit/mock/editable-storage-logic.spec.ts [hunk 4]
```diff
@@ -17,130 +17,278 @@ describe('Mock: Editable storage logic', () => {
...
+      mockCounter = await counterFactory.deploy(1);
+
+      await mockCounter.setVariables({
+        deployer: ADDRESS_EXAMPLE,
+        count: 3,
+      });
+
+      expect(await mockCounter.deployer()).to.equal(ADDRESS_EXAMPLE);
+      expect(await mockCounter.count()).to.equal(3);
+    });
```
Comment (wei3erHase) [inline test/unit/mock/editable-storage-logic.spec.ts:247]: i'd remove this one and the `Counter` types and factory imports!
the next text is good enough

**Comment by wei3erHase** [comment]
please add yourself as contributor here: https://app.dework.xyz/i/0FDGv5pCT7TJXZVki1wJ0Z

**Comment by tonykogias** [comment]
> please add yourself as contributor here: https://app.dework.xyz/i/0FDGv5pCT7TJXZVki1wJ0Z

removed the test and assigned myself on the link.

**Review by wei3erHase — APPROVED** [review]
this one looks ready to go! 🚀 
thanks for your contribution! 🦾 
we're pushing it to `dev` as we expect some new features to be added soon, and can make a canary release so that you early integrate it to your tests 👏

**Comment by wei3erHase** [comment]
cannary release at @defi-wonderland/smock@0.0.0-72aec142 `<3`

**Pull request merged**